okio/src/hashFunctions/kotlin/okio/internal/Hmac.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio.internal
+
+import okio.ByteString
+import kotlin.experimental.xor
```

Comment:
We can't rely on this in the main source set. I think Jesse wrote an internal version somewhere...

Reply:
I found this which seems to be the idea:

https://github.com/square/okhttp/blob/19771365f25d986d125478a75c15d24d6e285e12/okhttp/src/main/kotlin/okhttp3/internal/ws/WebSocketProtocol.kt#L109

Which, from what I can tell, basically comes down to inlining `kotlin.experimental.xor`

```kotlin
public inline infix fun Byte.xor(other: Byte): Byte = (this.toInt() xor other.toInt()).toByte()
```

I guess we want to do that to avoid compatibility changes when those methods leave the experimental stage?

Reply:
Replaced with internal copy of method